src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -66,15 +72,16 @@ private protected override INameResolver CreateResolver(PowerFxConfig alternateC
         /// <inheritdoc/>
         protected override IExpression CreateEvaluator(CheckResult result)
         {
-            return CreateEvaluatorDirect(result);
+            return CreateEvaluatorDirect(result, GetMaxCallDepth());
         }
 
         /// <summary>
         /// Create an evaluator over the existing binding. 
         /// </summary>
         /// <param name="result">A successful binding from a previous call to <see cref="Engine.Check(string, RecordType, ParserOptions)"/>. </param>
+        /// <param name="maxCallDepth">Max number of function nesting calls.</param>
         /// <returns></returns>
-        public static IExpression CreateEvaluatorDirect(CheckResult result)
+        public static IExpression CreateEvaluatorDirect(CheckResult result, int maxCallDepth = 100)

```

Comment:
>int maxCallDepth = 100 [](http://example.com/codeflow?start=76&length=22)

Adding an optional param to a public method  is a breaking change, as we noted recently from the PROSE changes. Is this necessary, and if so, should CreateEvaluatorDirect just take a powerfxconfig with this set? Assuming we'd want to have other config items like this eventually.

Reply:
Done

Reply:
What changed here? Is this the right approach over a config object?

Reply:
Unneeded, recursion depth handled in seperate PR.